coderd/httpmw/apikey.go

```diff
@@ -258,7 +269,7 @@ func ExtractAPIKey(rw http.ResponseWriter, r *http.Request, cfg ExtractAPIKeyCon
 			})
 		}
 		// Check if the OAuth token is expired
-		if link.OAuthExpiry.Before(now) && !link.OAuthExpiry.IsZero() && link.OAuthRefreshToken != "" {
+		if !link.OAuthExpiry.IsZero() && link.OAuthExpiry.Before(now) {
```

Comment:
Pondering: Not new behavior, but there doesn't seem to be any buffer between expiry and renewal (e.g. now - 10s), could this lead to bad user experiences where we approve the token and it expires right after?

Reply:
It should be ok, because refresh tokens are still valid when the oauth code expires. So we do not need to be preemptive here